src/tsolvers/lasolver/Simplex.cpp

```diff
@@ -0,0 +1,490 @@
+//
+// Created by prova on 06.09.19.
+//
+
+#include "Simplex.h"
+
+// MB: helper functions
+namespace{
+    bool isBoundSatisfied(Delta const & val, LABound const & bound ) {
+        if (bound.getType() == bound_u){
+            return val <= bound.getValue();
+        }
+        else {
+            assert(bound.getType() == bound_l);
+            return val >= bound.getValue();
+        }
+    }
+}
+
+bool Simplex::checkSimplex(std::vector<LABoundRef> &explanation, std::vector<opensmt::Real> &explanationCoefficients) {
```

Comment:
I don't like these out parameters that may or (more often) may not be used.
I think the simplex should just store the flag about the conflicting row and the solver would explicitly ask for the explanations.

Reply:
I'm not sure if I understand this one. The explanation especially in Simplex should have a limited lifetime that is specific to the query, so having it not live longer than necessary should be good.  The fact that the explanationCoefficients is not used is another issue that should be solvable by having perhaps an interpolating derived class of LASolver.

For the LRASolver I see that the explanation might need to have a longer lifetime, but I think this implementation did not change.

Reply:
Perhaps turning the thing into a return value as suggested below would fix the problem.

Reply:
Done.